web/src/main/java/com/bazaarvoice/emodb/web/resources/sor/DataStoreResource1.java

```diff
@@ -377,14 +374,22 @@ public Table getTableMetadata(@PathParam ("table") String table) {
             response = Iterator.class
     )
     @ApiImplicitParams ({@ApiImplicitParam (name = "APIKey", required = true, dataType = "string", paramType = "query")})
-    public Iterator<Map<String, Object>> scan(@PathParam ("table") String table,
+    public Object scan(@PathParam ("table") String table,
                                               @QueryParam ("from") String fromKeyExclusive,
                                               @QueryParam ("limit") @DefaultValue ("10") LongParam limit,
+                                              @QueryParam ("includeDeletes") @DefaultValue("false") BooleanParam includeDeletes,
                                               @QueryParam ("consistency") @DefaultValue ("STRONG") ReadConsistencyParam consistency,
                                               @QueryParam ("debug") BooleanParam debug) {
-        return streamingIterator(
-                _dataStore.scan(table, Strings.emptyToNull(fromKeyExclusive), limit.get(), consistency.get()),
-                debug);
+        // Always get all content, including deletes, from the backend.  That way long streams of deleted content don't
+        // create long pauses in results.
+        Iterator<Map<String, Object>> unfilteredContent;
+        if (includeDeletes.get()) {
+            unfilteredContent = _dataStore.scan(table, Strings.emptyToNull(fromKeyExclusive), limit.get(), true, consistency.get());
+            return streamingIterator(unfilteredContent, debug);
+        }
+        // Can't pass limit parameter to the back-end since we may exclude deleted content.  Get all records and self-limit.
+        unfilteredContent = _dataStore.scan(table, Strings.emptyToNull(fromKeyExclusive), Long.MAX_VALUE, true, consistency.get());
+        return deletedContentFilteringStream(unfilteredContent, limit.get());
```

Comment:
The control flow is funky here. Recommend:
```java
        if (includeDeletes.get()) {
            Iterator<Map<String, Object>> unfilteredContent = _dataStore.scan(table, Strings.emptyToNull(fromKeyExclusive), limit.get(), true, consistency.get());
            return streamingIterator(unfilteredContent, debug);
        } else {
            // Can't pass limit parameter to the back-end since we may exclude deleted content.  Get all records and self-limit.
            Iterator<Map<String, Object>> unfilteredContent = _dataStore.scan(table, Strings.emptyToNull(fromKeyExclusive), Long.MAX_VALUE, true, consistency.get());
            return deletedContentFilteringStream(unfilteredContent, limit.get());
        }
```